### ISSUE TITLE
UI table content refresh, how to refresh filters also

### ISSUE THREAD

**Issue by ssocratous71** [abap2UI5/abap2UI5#705]
Hi,
  if we have an UI table and apply filters on it, but hen refresh(rebind) data to it, filters/sorting icon are still there but not applied. Is there a way to reapply current filters? (i.e. like in javascript UITable.applyFilters())

Regards,

S.Socratous

**Comment by oblomov-dev** [comment]
yes we can call frontend functions in abap2UI5,, for example on after rendering here:
https://github.com/abap2UI5/abap2UI5/blob/574802e9a7ea0f4f0c6e041c3607afc5c6186a60/src/01/01/z2ui5_cl_fw_http_handler.clas.abap#L89

Is it pssible that you create a small example app with you use case? you can let me know then which function you need to call after the model update and i can try to find a good place to add this to abap2UI5.

**Comment by ssocratous71** [comment]
As said above, I need a way to reapply the current filters after a model update on a ui table. The sapui5 command is something like oTable.getBinding("items").applyFilter(); Also, it will be good to have a clear all filters kind of functionaliy in the toolbar: sample code for this can be found in:
https://sapui5.hana.ondemand.com/#/entity/sap.ui.table.Table/sample/sap.ui.table.sample.Filtering/code

Thanks again.

**Comment by oblomov-dev** [comment]
yes but there is a lot more happing at the frontend when you check the controller file. Are you sure that the filtering is running out of the box? I only did filtering with sap.m.table and made everything in the backend.
Z2UI5_CL_DEMO_APP_059

With demo i mean creating an app in abap2UI5 and adding as far as possible all functionality of the sample [(here)](https://sapui5.hana.ondemand.com/#/entity/sap.ui.table.Table/sample/sap.ui.table.sample.Filtering/code) which you need. Then i can save the time to create the demo by myself and just can take a look to the missing functionality and check if there is a way to add this to abap2Ui5.

**Comment by ssocratous71** [comment]
OK, I will build a demo app with table filtering and send it here.

**Comment by oblomov-dev** [comment]
Added now the functionality that filters are reset after a model update to abap2UI5.

you can check the z2ui5_cl_demo_app_143 example:
https://github.com/abap2UI5/abap2UI5-samples/blob/d821ff9ae78fac2c64e1e9be9edb3eebe6ec4bd2/src/z2ui5_cl_demo_app_143.clas.abap#L69

![gif_ui_filter](https://github.com/abap2UI5/abap2UI5/assets/102328295/8c94034f-1b41-4bc9-9848-86bf470fb837)

The function is encapsulated in a custom control:
https://github.com/abap2UI5/abap2UI5/blob/main/src/01/03/z2ui5_cl_fw_cc_uitableext.clas.abap

This custom control can be improved and is just my fast solution to find a way where this functionality can be added. For example it uses a timer instead of attaching to the onafter rendering event etc. but see it as a starting point. and you can add more functionality with:
https://blogs.sap.com/2023/12/11/abap2ui5-a4-extensions-ii-guideline-for-developing-new-features-in-javascript/

Best regards.

**Comment by oblomov-dev** [comment]
i close this issue, just open a new one if you have question or need further functionality.
